### PR TITLE
First shot at babelifying the server

### PR DESCRIPTION
**Pull request by phillipj**
Started slowly with index.js / server.js, can transition the rest later.

Refs #26.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -28,17 +40,11 @@ hbs.registerHelper('__n', function () {
...
 });
 
-var serverConnection = server.connection({
-  port: config.get('port'),
-  router: {
-    stripTrailingSlash: true
-  }
-});
 server.register({
-  register: require('good'),
```
Comment (phillipj) [inline server.js]: AFAIK we cant do inline requiring when using `import`

Comment (leftieFriele) [inline server.js]: Not without using the [require hook](https://babeljs.io/docs/usage/require/), no.

Comment (phillipj) [inline server.js]: That is being used in `index.js` to bootstrap it all. I'm talking about doing inline require within the code like it used to on line #38 isnt possible when transitioning to harmony style imports. E.g.
```javascript
server.register({
  register: import good from 'good'
});
```
I have tried different approaches to accomplish this before without luck.

Comment (leftieFriele) [inline server.js]: This is totally fine imho :+1

Comment (phillipj) [inline server.js]: w000p :dancers:

**Comment by phillipj** [comment]
Forgot to mention this also includes running and testing with node v4.

**Pull request merged**